crates/oxc_codegen/src/lib.rs

```diff
@@ -590,4 +607,27 @@ impl<'a> Codegen<'a> {
     fn try_take_moved_comment(&mut self, node_start: u32) -> Option<Vec<AnnotationComment>> {
         self.move_comment_map.remove(&node_start)
     }
+
+    fn print_leading_comments(&mut self, start: u32) {
+        if self.options.minify {
+            return;
+        }
+        let Some(source_text) = self.source_text else { return };
+        let Some(comments) = self.leading_comments.remove(&start) else { return };
```

Comment:
Performance killer :-/ we need to somehow carry over whether a token has leading comments to the ast node, for a fast path.